core/src/main/java/org/apache/accumulo/core/iterators/SleepyIterator.java

```diff
@@ -0,0 +1,80 @@
+/*
```

Comment:
There is a similar functioning iterator with org.apache.accumulo.test.functional.SlowIterator.  Assuming that this iterator is used only for testing, would a test package be a better fit?

Also, many of the tests that use SlowIterator are flaky - beloved to be because of resource contention and those tests usually needing multiple, concurrent operations.  

- Would sleep iterator help with those tests?
- Are tests that use sleepy iterator likely to have similar issues?

Reply:
I will look at moving this to a test package. For clarification, this iterator is only used in tests that are trying to make a compaction take long enough in MAC that the test thread can cancel or delete it. The iterator is not used in the main test thread, so I don't think it should make the test flaky. I'll have to look at SlowIterator to see how/where it's used and if SleepyIterator can be used in it's place or vice versa.

Reply:
@EdColeman  - thanks for pointing out SlowIterator. I went ahead and just removed my iterator instead.

Reply:
I was hoping yours was an improvement :-(

Reply:
lol - no, looking at SlowIterator, mine duplicated it.